packages/rlc-common/src/metadata/packageJson/azurePackageCommon.ts

```diff
@@ -31,10 +31,13 @@ export function getAzureCommonPackageInfo(config: AzurePackageInfoConfig) {
 export function getAzurePackageDependencies({
   hasLro,
   hasPaging,
-  specSource
+  specSource,
+  isModularLibrary
 }: AzurePackageInfoConfig) {
   let dependencies: Record<string, string> = {
-    "@azure-rest/core-client": specSource === "Swagger" ? "^1.4.0" : "^2.0.0",
+    // Swagger and modular libraries currently only support the old multipart/form-data implementation.
+    "@azure-rest/core-client":
+      specSource === "Swagger" || isModularLibrary ? "^1.4.0" : "^2.0.0",
```

Comment:
why modular is depend on v1?

Reply:
For modular to depend on v2, we need to update the modular API models as well, and probably also figure out how we want to handle serialization, which is a whole other piece of work which is I think out of scope for this PR. So I kept modular on the old version for now. Obviously we aren't going to keep it like that.

Reply:
> For modular to depend on v2, we need to update the modular API models as well, and probably also figure out how we want to handle serialization

Personally I prefer to always use the latest version in modular. We didn't support mfd feature in core rest v1 either, it's fine considering there is no real need to modular mfd yet except openai(but i heard that we plan to depreate azure openai ) AND modular is still in preview stage.  The version mismatch may introduce un-necessary maintainance effort.

Reply:
I think it depends, if making modular take a dependency on v2 means there is additional effort in this PR I would favor keeping it in v1. If it is just a matter fo switching the string, go for it. However I think it is most likely the former, if so I'm fine keeping the 2 versions in the meantime.

Reply:
ah, I think I see what you mean. I was under the impression that MFD was somewhat working in modular v1 given OpenAI is using it. But that is because they are doing customization, right? In that case I will change this to use 2.0, since there won't be a regression (it was already broken).

At the moment I have modular just generating normal models for MFD so as not to break the compilation.

Reply:
yes, openai has customization here.